status/manager.go

```diff
@@ -78,23 +110,27 @@ func (s *Manager) SubServerStatus(_ context.Context,
 
 // RegisterSubServer will create a new sub-server entry for the Manager to
 // keep track of.
-func (s *Manager) RegisterSubServer(name string) {
+func (s *Manager) RegisterSubServer(name string,
+	overrides map[string]func() bool) {
+
 	s.mu.RLock()
 	defer s.mu.RUnlock()
 
-	ss := newSubServerStatus()
+	ss := newSubServerStatus(overrides)
 	ss.Status = "Registered"
 
 	s.subServers[name] = ss
 }
 
 // RegisterAndEnableSubServer will create a new sub-server entry for the
 // Manager to keep track of and will set it as enabled.
-func (s *Manager) RegisterAndEnableSubServer(name string) {
+func (s *Manager) RegisterAndEnableSubServer(name string,
+	overrides map[string]func() bool) {
+
 	s.mu.RLock()
 	defer s.mu.RUnlock()
 
-	ss := newSubServerStatus()
+	ss := newSubServerStatus(overrides)
```

Comment:
btw - I think we should remove the `GetStatus` method below & unexport the `SubServerStatus` struct (and probably just rename it to `subServer`) since we dont want to expose the values of the struct to outside this package.

Reply:
Good idea, addressed with the latest push!